core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.spec.ts

```diff
@@ -581,6 +633,36 @@ describe('OptimizedSsrEngine', () => {
         expect(engineRunner.renders).toEqual(['', 'a-1']);
       }));
     });
+
+    it('should render fallback to CSR when rendering request fails', fakeAsync(() => {
+      const engineRunner = new TestEngineRunner(
+        {
+          renderingStrategyResolver: () => RenderingStrategy.ALWAYS_SSR,
+          timeout: 50,
+        },
+        defaultRenderTime,
+        true
+      );
+      engineRunner.request('a');
+      tick(200);
+      expect(engineRunner.renders).toEqual([new Error('a-0')]);
+    }));
+
+    it('should render fallback to CSR when followup request fails', fakeAsync(() => {
+      const engineRunner = new TestEngineRunner(
+        {
+          renderingStrategyResolver: () => RenderingStrategy.ALWAYS_SSR,
+          timeout: 50,
+        },
+        defaultRenderTime,
+        true
+      );
+      engineRunner.request('a');
+      tick(200);
+      engineRunner.request('a');
+      tick(200);
+      expect(engineRunner.renders).toEqual([new Error('a-0'), new Error('a-1')]);
+    }));
```

Comment:
This test is very similar to the test above. I have hard time understanding the value of this spec scenario with 2 requests, in the light of ALWAYS_SSR and timeout being ignored.
My first thought was that maybe we should remove this spec. But I might be missing something important